### PR TITLE
feat(log): add logout and ping timeout log

### PR DESCRIPTION
**Pull request by utkarshmani1997**
This commit add log for the ping  request timeout from initiator side, and for logout request.
Signed-off-by: Utkarsh Mani Tripathi <utkarsh.tripathi@mayadata.io>

### CODE REVIEW

File: pkg/port/iscsit/iscsid.go [hunk 1]
```diff
@@ -466,6 +471,7 @@ func (s *ISCSITargetDriver) iscsiExecLogin(conn *iscsiConnection) error {
 }
 
 func iscsiExecLogout(conn *iscsiConnection) error {
+	log.Infof("Logout request received from initiator: %q, address: %v", conn.session.Initiator, conn.conn.RemoteAddr().String())
```
Comment (vishnuitta) [inline pkg/port/iscsit/iscsid.go]: session can be null here.. we can't do this

File: pkg/port/iscsit/iscsid.go [hunk 2]
```diff
@@ -873,6 +880,10 @@ func (s *ISCSITargetDriver) scsiCommandHandler(conn *iscsiConnection) (err error
 			conn.session.PendingTasksMutex.Unlock()
 		}
 	case OpNoopOut:
+		timeout := time.Since(conn.lastNopout)
+		if timeout > opPingTimeout {
+			log.Warningf("Received ping after %v, ping timeout is %v", timeout, opPingTimeout)
+		}
 		iscsiExecNoopOut(conn)
```
Comment (vishnuitta) [inline pkg/port/iscsit/iscsid.go]: we need to check the time taken by iscsiExecNoopOut function call. We don't need to check the time from last nopout.
Its a good idea to track it, but, doing this here won't help and the same need to be done in a long-time running thread. Lets take this in another PR.
For this PR, just find the time taken to execute 'iscsiExecNoopOut'. And, do a warning log if it takes more than 3 seconds

File: pkg/port/iscsit/session.go [hunk 3]
```diff
@@ -389,10 +389,11 @@ func (s *ISCSITargetDriver) BindISCSISession(conn *iscsiConnection) error {
 		}
 
 		if newSess.SessionType == SESSION_NORMAL {
-			log.Infof("New Session, type: %s, initiator name:%v, target name:%v, ISID:0x%x",
-				"Normal", conn.loginParam.initiator, conn.loginParam.target, conn.loginParam.isid)
```
Comment (vishnuitta) [inline pkg/port/iscsit/session.go]: isid also will be good to have

File: pkg/port/iscsit/iscsid.go (hunk 1 above)
Comment (utkarshmani1997) [reply pkg/port/iscsit/iscsid.go]: ok

File: pkg/port/iscsit/iscsid.go (hunk 2 above)
Comment (utkarshmani1997) [reply pkg/port/iscsit/iscsid.go]: ok

File: pkg/port/iscsit/session.go (hunk 3 above)
Comment (utkarshmani1997) [reply pkg/port/iscsit/session.go]: ok

File: pkg/port/iscsit/iscsid.go [hunk 4]
```diff
@@ -87,6 +89,10 @@ const (
 	STATE_TERMINATE
 )
 
+var (
+	opPingTimeout = 3 * time.Second
```
Comment (vishnuitta) [inline pkg/port/iscsit/iscsid.go]: can you change it to `opPingThreshold`

**Comment by vishnuitta** [comment]
can you pick this commit to jiva @utkarshmani1997 ?

**Pull request merged**